### PR TITLE
#[rustc_error] survey

### LINKED ISSUE #55505: should survey uses of `#[rustc_error]` and see if they can be replaced with `// compile-pass`
Discussing some of the difficulties with #54528  with @nikomatsakis led them to remind me that we originally introduced `#[rustc_error]` because we didn't *have* a way to express `// compile-pass` to the test infrastructure.

But now we do

and it would make sense for us to try to migrate instances of `#[rustc_error]` to use `// compile-pass` *when possible*.

(I think the main case where it is not possible are instances of `// revisions` where at least one of the revisions compiles successfully and another does not.)

### PR DESCRIPTION
**Pull request by pnkfelix**
Fix #55505

### CODE REVIEW

File: src/test/ui/asm/asm-misplaced-option.rs [hunk 1]
```diff
@@ -21,14 +21,14 @@
 // ignore-mips
 // ignore-mips64
 
-#![feature(asm, rustc_attrs)]
-
+// compile-pass
+// skip-codegen
```
Comment (ishitatsuyuki) [inline src/test/ui/asm/asm-misplaced-option.rs]: Wait, can this be tested without using any part of LLVM? That's what `skip-codegen` do.

Comment (pnkfelix) [reply src/test/ui/asm/asm-misplaced-option.rs]: We still get the warnings encoded in asm-misplaced-option.stderr while doing skip-codegen 

Is there something else you expect to be tested here?

Comment (ishitatsuyuki) [reply src/test/ui/asm/asm-misplaced-option.rs]: I didn't expect rustc to have an asm parser (and I thought that was done by LLVM), but I guess I'm wrong.

Comment (pnkfelix) [reply src/test/ui/asm/asm-misplaced-option.rs]: ... I think this test is *just* testing an aspect of how the macro arguments are being fed in.
I don't think there's any testing of the actual contents of the asm block.

Note in particular that this test was previously using `#[rustc_error]`, which means we've *never* been feeding it into LLVM; its always been erroring before code generation would start.

Comment (ishitatsuyuki) [reply src/test/ui/asm/asm-misplaced-option.rs]: Ah OK, that makes sense.

**Comment by kennytm** [comment]
The bors queue has a spurious failure (30 min timeout which is likely related to 9696), but the CI PR reported in https://github.com/rust-lang/rust/pull/55532#issuecomment-435855500 looks legit, so I'm not going to retry.

**Comment by pnkfelix** [comment]
That is very goofy. I do not see that test result locally. Is there any chance that our diagnostic here is behaving differently on different platforms...?

**Comment by pnkfelix** [comment]
(isn't it also weird that the travis result says this:

```
[00:50:27] failures:
[00:50:27]     [ui] ui/lint/lint-type-overflow2.rs
[00:50:27]     [ui] ui/lint/lint-type-overflow2.rs
```

Why is that occurring twice?

**Comment by pnkfelix** [comment]
Oh, the raw log has it only once.

So that's something wrong with some other piece of post-processing, I guess

**Comment by pnkfelix** [comment]
okay ... at this point I assume this must be some weird artifact of my own environment, maybe some customization in my `config.toml`. Looking.

**Comment by pnkfelix** [comment]
(ah  it might be due to an `optimize-tests = false` in my `config.toml`...

**Comment by pnkfelix** [comment]
Yes, okay, the presence/absence of `-O` causes the lint output to differ here.

**Comment by oli-obk** [comment]
I usually add `// compile-flags: -O` to such tests, so that `optimize-tests` does not change the behavior`

**Pull request merged**